### DISCUSSION TITLE
Rule suddenly not alerting

### DISCUSSION THREAD

**Discussion by lymanaukuleles** [jertel/elastalert2#856] [Q&A] [answered]
I have no idea why (or what I did) but currently one of my rule files is not alerting. There doesn't appear to be any Error messages related to it in my `docker logs`. Also it is not in the `Disabled Rules []`. The query is the same as it has always been. So I'm confused. It is running fine but just not picking up on query hits or alerting. My other rule files are working fine.

**Comment by jertel** [comment, marked as answer]
You will need to enable debug logs and closely review the query being made to Elasticsearch and the response coming back. Compare those with your equivalent queries in Kibana (using the Inspect tab) to see why Kibana finds the data but not ElastAlert 2.

**Reply by lymanaukuleles** [reply]
This is what I'm getting, unfortunately I can't see a hit/match because it isn't getting one:
```
2022-05-21 15:32:44,057    DEBUG        elasticsearch > {"query":{"bool":{"filter":{"bool":{"must":[{"range":{"@timestamp":{"gt":"2022-05-21T15:17:44.046134Z","lte":"2022-05-21T15:32:44.046134Z"}}},{"query_string":{"query":"\"LDAP authentication failure\""}}]}}}},"sort":[{"@timestamp":{"order":"asc"}}]}
2022-05-21 15:32:44,058    DEBUG        elasticsearch < {"took":0,"timed_out":false,"_shards":{"total":0,"successful":0,"skipped":0,"failed":0},"hits":{"total":{"value":0,"relation":"eq"},"max_score":0.0,"hits":[]}}
```

**Reply by jertel** [reply]
How do you know those "LDAP authentication failure" logs are still actually being stored in Elasticsearch?

**Reply by lymanaukuleles** [reply]
Because I can search for and view them in my Kibana. Does that prove that they are being stored in Elasticsearch?

**Reply by jertel** [reply]
Yes. I asked you in my first response to compare the ElastAlert 2 debug request and response (which you then posted) to the Kibana requests/response (via the Inspect tab) but you never commented on the Kibana request/response values. You have to follow through with that part of the troubleshooting process.

**Reply by lymanaukuleles** [reply]
Here is a log from Kibana (sorry kind of weird formatting):

```
_id | bKKk6oABInn_j6EKsC74
-- | --
_index | logstash
_score | 1
_type | _doc
@timestamp | May 21, 2022 @ 21:21:29.000
@version | 1
c | 16
dst | 128.181.103.2:0:X1
fw | 128.181.103.2
fw_action | NA
gcat | 4
host.name | 10.100.10.4
id | firewall
m | 745
msg | User login denied - LDAP authentication failure
n | 747
note | john
pri | 6
proto | tcp
received_at | May 21, 2022 @ 21:21:42.987
received_from | 10.100.10.4
sess | sslvpnc
sn | 18B169961D00
src | 141.239.215.245:0:X1
syslog.priority | 46
tags | ecs_converted
time | 2022-05-21 21:21:29
type | syslog
usr | john@whtasi.local
```
So I still don't see the issue?

**Reply by jertel** [reply]
Is your rule using the `logstash` index?

If you type in "LDAP authentication failure" into the Kibana search bar do you get the logs?

**Reply by lymanaukuleles** [reply]
Yes to both your questions.

Once again the rule and alert was working fine until about a week ago. I have no idea what caused it to stop. I also tried deleting and recreating the rule file, as well as creating a completely new docker container, but same issue.

EDIT: Actually I think I found the issue. I changed my logstash index pattern from daily indices to just logstash. This changed the template from logstash-* to just logstash*. My rule file is searching for logstash-* not logstash*.

**Reply by jertel** [reply]
Grab the Kibana "Inspect" request/response pairs from when you search for "LDAP authentication failure". I've mentioned this twice already. You need those to compare to the ElastAlert 2 request/response pairs.
